### PR TITLE
Remove get_filters() from controller

### PR DESCRIPTION
**Pull request by kmws**
(This is a part of #213)

### CODE REVIEW

File: include/view.h [hunk 1]
```diff
@@ -173,6 +174,7 @@ class view {
 	std::string last_fragment;
 	unsigned int tab_count;
 	cache* rsscache;
+	filtercontainer* filters;
```
Comment (Minoru) [inline include/view.h]: Please initialize this field to `nullptr`.

**Comment by kmws** [comment]
Thank you for the reviews, @Minoru. After #301 is done and merged, I will rebase both my PRs (#302 and #304) and fix the initialization thing in this one.

**Comment by Minoru** [comment]
#301 is merged, you can rebase now.

**Comment by Minoru** [comment]
A lot of files were renamed, and you probably have old object files still lying around. If so, you can do one of the following:
- check out some older commit (from before #301 got merged) and run `make distclean`;
- or just run `find -name '*.o' -delete`. That always worked for me so far, but is a bit hacky IMHO.

**Comment by kmws** [comment]
>or just run find -name '*.o' -delete

That did the rick... thank you!

**Comment by kmws** [comment]
Hmm. I've tried to `git pull --rebase`, but the result is pretty bad. Should I just open new PR with cleanly applied changes?

**Comment by der-lyse** [comment]
Not sure if it's the best way, but you can try the following: branch off of current master and then cherry-pick your last two commits of this branch with the actual work and than force-push to this branch here.

Alternatively just do `git rebase master` on this branch, however I'm not sure whether this will work in this case or not. Good luck! ;-)

(In order to combine both of the commits into a single one, I'd do an interactive rebase (`git rebase -i HEAD~2`) and then `squash` or just `fixup` the second into the first one.)

**Comment by der-lyse** [comment]
Eh, hang on, I just took a look at the two "main" commits and they don't seem to do what the commit message states they do. So I'm not sure whether redoing it is easier or cherry-picking the original commits and fixing the resulting merge conflicts.

**Comment by kmws** [comment]
>branch off of current master and then cherry-pick your last two commits of this branch with the actual work and than force-push to this branch here.

Well I tried this, but if I try to force-push it says _Everything up to date._ :-(

**Comment by Minoru** [comment]
I'm not sure what went wrong, but it looks like you rebased onto zaowen's branch rather than master, because I don't see my merge commit in your branch. If you're getting "Everything up to date" despite retrying, your `master` must be old. Check it out once more, do a pull, and make sure it's now on commit 8069b1fb. After that, retry Lyse's suggestion about cherry-picking.

If none of this helps, I can go in and fix this myself ([GitHub lets me change your branch](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/)), so you don't have to create a new PR. But I'd rather help you figure this out if you're in the mood ;)

**Comment by kmws** [comment]
>it looks like you rebased onto zaowen's branch rather than master

No, I pulled from Newsboat's `master` (as always) before branching out (in an effort to force-push later). My local master branch is indeed at `8069b1f`. Then I did all of my changes (as one commit) and tried to force-push here, but it says it's up to date...

First I tried to git pull --rebase from Newsboat's `master`, resolved some conflicts, did `rebase --continue` and my changed were looking like they are on top of your merge, which seemed OK. Huh. :-(

**Comment by Minoru** [comment]
> No, I pulled from Newsboat's master (as always) before branching out

Yeah; it'd be hard to rebase onto someone else's PR without noticing :) But I just don't have any other explanation for the current state of history in this PR.

> My local master branch is indeed at 8069b1f.

That's reassuring, I think we can fix this still (**not** the [xkcd way](https://xkcd.com/1597/), but for real)! 

> Then I did all of my changes (as one commit) and tried to force-push here, but it says it's up to date...

Can you re-try that now, and post what commands you ran and what output you got? There's probably some small detail that we're missing, maybe the output will hint at it.

**Comment by kmws** [comment]
```
$ git checkout get-filters-2
Switched to branch 'get-filters-2'
$ git log --oneline -n 2
c6abaaf1 (HEAD -> get-filters-2) Remove get_filters() from controller
8069b1fb (gh/master, master) Merge pull request #301 from zaowen/ClassRename
$ git pull gh master
From https://github.com/newsboat/newsboat
 * branch              master     -> FETCH_HEAD
 Already up to date.
$ git push origin get-filters
Everything up-to-date

```
`c6abaaf1` contains all of the changes as one commit.

**Comment by Minoru** [comment]
Ah, here's the detail I was suspecting:
```shell
$ git push origin get-filters
```

This command pushes branch `get-filters` to its remote branch in `origin`, which is `get-filters` by default. It does so even though you're on branch `get-filters-2` right now. Run this instead:

```shell
$ git push origin --force-push get-filters-2:get-filters
```

The last argument means "push local branch `get-filters-2` to remote branch `get-filters`".

**Comment by kmws** [comment]
Wows. Thank you, that was it!
It seems that some of the `clang-format` changes sneaked in. If they are not OK, I can undo them and force again (to train ;).

**Comment by Minoru** [comment]
That's Git for you: it's flawless when it works, and it's a total puzzle when it doesn't :)

The clang-format changes seem fine, but please put them into a separate commit. A good opportunity to try out `git reset --soft "HEAD~"` and `git add --patch` if you haven't already ;)

**Comment by Minoru** [comment]
Coveralls went mad from all that rebasing :)

Thank you for the contribution, @kmws! Don't forget to rebase your other PRs ;)

**Comment by kmws** [comment]
Once again, thank you for taking the time to help me, @Minoru. And thank you too, @der-lyse, we were close. ;)

**Pull request merged**